model/dependency.go

```diff
@@ -75,15 +76,29 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 		return included, nil
 	}
 
-	// if the given task is a task group, recurse on each task
+	// For a task group, recurse on each task and add those tasks that should be
+	// included.
 	if tg := di.Project.FindTaskGroup(pair.TaskName); tg != nil {
+		catcher := grip.NewBasicCatcher()
 		for _, t := range tg.Tasks {
 			ok, err := di.handle(TVPair{TaskName: t, Variant: pair.Variant}, activationInfo, generatedVariants, false)
-			if !ok {
-				di.included[pair] = false
```

Comment:
can we keep this line so we don't have to check unschedulable tasks over and over again?

Reply:
I read over the logic and I believe that setting `di.included[pair] = false` in every `!ok` case from `di.handle` doesn't actually do anything because it already sets the task inclusion false in every return case where `di.handle` returns false ([here](https://github.com/evergreen-ci/evergreen/blob/86e8b8340c5c7eb8a65575d6797613fcda59c5c2/model/dependency.go#L92-L107)).

Regardless of whether `ok` is used or not, I don't think the task group itself (which is represented by `pair`) needs to be added to `di.included` since it'll never be true. The task group is just a logical grouping mechanism, so it's not a real task you can schedule or depend on (kind of like an execution task is a real task and a display task is just a different way of showing it in the UI).